### PR TITLE
[USER32] Fix caret coordinates and clipping

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Fix caret coordinates and clipping.
JIRA issue: [CORE-10201](https://jira.reactos.org/browse/CORE-10201)

### CODE REVIEW

File: win32ss/user/ntuser/caret.c [hunk 1]
```diff
@@ -29,17 +31,21 @@ co_IntDrawCaret(PWND pWnd, PTHRDCARETINFO CaretInfo)
...
     }
 
-    if (pWnd->hrgnUpdate)
-    {
-       NtGdiSaveDC(hdc);
-    }
+    NtGdiSaveDC(hdc);
+
+    IntGetClientRect(pWnd, &rcClient);
+    NtGdiIntersectClipRect(hdc,
```
Comment (tkreuzer) [inline win32ss/user/ntuser/caret.c]: After removing DCX_WINDOW this should not be required anymore. On the other hand, when I look at the implementation of UserGetDCEx, it seems like it doesn't set the client region as clip region. So to me it looks like the bug is there. Can you investigate that?

Comment (katahiromz) [reply win32ss/user/ntuser/caret.c]: See https://jira.reactos.org/browse/CORE-10201 .
I confirmed that GetDC and GetDCEx(hwnd, NULL, DCX_CACHE) enable clipping.

**Comment by katahiromz** [comment]
Done.

**Comment by katahiromz** [comment]
Try again.

**Comment by JoachimHenze** [comment]
tkreuzer, Seems as if you had some objections here, I wanted to inform you that this PR got superseded by PR1271, which then got committed into 0.4.12-dev-401-gefe4605 (without any review as it seems).
So if you want you could quickly double-check whether your points got addressed within PR1271.
My practical testing for PR1271 gave good results.

**Pull request closed without merging**